cmd/kubeadm/app/cmd/phases/selfhosting.go

```diff
@@ -48,5 +57,8 @@ func NewCmdSelfhosting() *cobra.Command {
 	}
 
 	cmd.Flags().StringVar(&kubeConfigFile, "kubeconfig", "/etc/kubernetes/admin.conf", "The KubeConfig file to use for talking to the cluster")
+	cmd.Flags().StringVar(&featureFlagsString, "feature-gates", featureFlagsString, "A set of key=value pairs that describe feature gates for alpha/experimental features. "+
```

Comment:
What about a helper func that would look something like this? Not sure how fancy we wanna get, but:
```go
// in features.go
type ChosenFeatures struct {
     featureGatesString *string
}

func AddFeaturesToFlagSet(fs *pflag.FlagSet, featureList *FeatureList) *ChosenFeatures {
    cf := &ChosenFeatures{}
    fs.StringVar(cf.featureGatesString, "feature-gates", "", "A set of key=value pairs that describe feature gates for alpha/experimental features. Options are:\n"+strings.Join(features.KnownFeatures(&features.InitFeatureGates), "\n"))
    return cf
}

func (cf *ChosenFeatures) GetFeaturesEnabledMap() (map[string]bool, error) {
    // use *cf.featureGatesString here
}
```

Then usage would be pretty straightforward:
```go
var enabledFeatures features.ChosenFeatures

// in cobra Run() {
    var err error
    cfg.FeatureGates, err = enabledFeatures.GetFeaturesEnabledMap()
    if err != nil {
         return err
    }
}

enabledFeatures := AddFeaturesToFlagSet(cmd.Flags(), &features.InitFeatureGates)
```

Not sure if this is a better or worse approach. I'll leave it up to you to decide.
We can always rework it later if we find the need for it.